src/sagemaker/djl_inference/model.py

```diff
@@ -0,0 +1,817 @@
+    """
+
+    def __init__(
+        self,
+        endpoint_name,
+        sagemaker_session=None,
+        serializer=JSONSerializer(),
+        deserializer=JSONDeserializer(),
+    ):
+        """Initialize a ``DJLLargeModelPredictor``
+
+
+        Args:
+            Args:
+            endpoint_name (str): The name of the endpoint to perform inference
+                on.
+            sagemaker_session (sagemaker.session.Session): Session object that
+                manages interactions with Amazon SageMaker APIs and any other
+                AWS services needed. If not specified, the estimator creates one
+                using the default AWS configuration chain.
+            serializer (sagemaker.serializers.BaseSerializer): Optional. Default
+                serializes input data to json format. Handles lists and numpy
+                arrays.
+            deserializer (sagemaker.deserializers.BaseDeserializer): Optional.
+                Default parses the response from json format to dictionary.
+        """
+        super(DJLLargeModelPredictor, self).__init__(
+            endpoint_name,
+            sagemaker_session,
+            serializer=serializer,
+            deserializer=deserializer,
+        )
+
+
+def _determine_engine_for_model(model_type: str, number_of_partitions: int, num_heads: int):
+    """Placeholder docstring"""
+
+    # Tensor Parallelism with DeepSpeed is only possible if n_heads can be split evenly across devices
+    if num_heads % number_of_partitions:
+        return HuggingFaceAccelerateModel
+    if model_type in defaults.DEEPSPEED_RECOMMENDED_ARCHITECTURES:
+        return DeepSpeedModel
+    return HuggingFaceAccelerateModel
+
+
+def _validate_engine_for_model_type(cls, model_type: str, number_of_partitions: int, num_heads: int):
+    """Placeholder docstring"""
+
+    if cls == DeepSpeedModel:
+        if model_type not in defaults.DEEPSPEED_SUPPORTED_ARCHITECTURES:
+            raise ValueError(
+                f"{model_type} is not supported by DeepSpeed. "
+                f"Supported model_types are {defaults.DEEPSPEED_SUPPORTED_ARCHITECTURES}"
+            )
+        if num_heads % number_of_partitions:
+            raise ValueError("The number of attention heads is not evenly divisible by the number of partitions."
+                             "Please set the number of partitions such that the number of attention heads can be"
+                             "evenly split across the partitions.")
+    return cls
+
+
+def _read_existing_serving_properties(directory: str):
+    """Placeholder docstring"""
+
+    serving_properties_path = os.path.join(directory, "serving.properties")
+    properties = {}
+    if os.path.exists(serving_properties_path):
+        with open(serving_properties_path, "r") as f:
+            for line in f:
+                key, val = line.split("=", 1)
+                properties[key] = val
+    return properties
+
+
+class DJLLargeModel(FrameworkModel):
```

Comment:
is this easily extensible to other model servers if we want to? `DJL` is hardcoded into the class names now, if it's not highly model server dependent, is it possible to have generic name?